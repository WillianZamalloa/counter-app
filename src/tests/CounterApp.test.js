import { shallow } from "enzyme";
import CounterApp from "../CounterApp";
import { render } from '@testing-library/react';


describe('Pruebas con el <CounterApp />', ()=>{
    test('debe de mostrar <CounterApp /> correctamente ', () => {        
        const wrapper = shallow(<CounterApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar el valor por defecto 100 ', () => {        
        const wrapper = shallow(<CounterApp value={100}/>);
        const counterText = wrapper.find('h2').text().trim();
       
        expect(counterText).toMatchSnapshot('100');
    });

    test('debe de incrementar el valor del boton +1 ', () => {        
        const wrapper = shallow(<CounterApp />);
        const btn1 = wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('101');

       //console.log(btn1.html());
        //expect(counterText).toMatchSnapshot('100');
    });
    




});