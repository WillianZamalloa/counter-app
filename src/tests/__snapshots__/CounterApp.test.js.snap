// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el <CounterApp /> debe de incrementar el valor del boton -1 : 102 1`] = `"101"`;

exports[`Pruebas con el <CounterApp /> debe de mostrar <CounterApp /> correctamente  1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;

exports[`Pruebas con el <CounterApp /> debe de mostrar el valor por defecto 100 : 100 1`] = `"100"`;
